### ISSUE TITLE
SDL utils segfault

### ISSUE THREAD

**Issue by SirDifferential** [wesnoth/wesnoth#4855]
### Game and System Information
Wesnoth 1.14.11 downloaded from the Wesnoth website on Windows 10, AMD Ryzen 7 1700 + AMD Radeon RX 580 driver 20.1.2

### Describe the bug
I was playing the undead campaign SotA mission Merfolk Revenge on turn 2 when the main screen started glitching. It looked like the renderbuffer wasn't reseted between frames and new frames were rendered on top of the old frame. Please see the screenshot. After this, the game crashed with assertion failure in `src/sdl/utils.cpp:2012`

### To Reproduce

Unknown. Happened only once.
![wesnoth_bug002](https://user-images.githubusercontent.com/494901/79641033-b2553a80-819d-11ea-9d2d-0e25cf62d2e2.png)

**Comment by gfgtdf** [comment]
Please upload the logfiles associated to this crash, this page describes how to find them https://forums.wesnoth.org/viewtopic.php?f=4&t=46166#w0windows .There will be multiple files, please select upload the one from the last start before the crash has happened (17:49) or just zip and attack all of them

**Comment by Pentarctagon** [comment]
For future reference, this issue has been reported and discussed quite a bit [here](https://forums.wesnoth.org/viewtopic.php?p=654371#p654371) as well.

**Comment by gfgtdf** [comment]
@SirDifferential  it'd be nice if you could upload the logfiles soon because we'd oth doesn't keep them forever so if you wait too long it might no longer be possible later

**Comment by SirDifferential** [comment]
I believe this is the right log:

```
Battle for Wesnoth v1.14.11
Started on Sat Apr 18 14:50:10 2020

Automatically found a possible data directory at D:\games\Battle for Wesnoth 1.14.11

Data directory:               D:\games\Battle for Wesnoth 1.14.11
User configuration directory: C:\Users\USER\Documents\My Games\Wesnoth1.14
User data directory:          C:\Users\USER\Documents\My Games\Wesnoth1.14
Cache directory:              C:\Users\USER\Documents\My Games\Wesnoth1.14\cache

Setting mode to 1920x1057
Checking lua scripts... ok
Detected an image with bad dimensions: items/gold-coins.png: 60x37
It will not be masked, please use: 72x72
20200418 17:38:56 error display: could not open image 'maps/background.jpg'
20200418 17:38:56 error gui/draw: Image: 'maps/background.jpg' not found and won't be drawn.
20200418 17:48:48 error display: could not open image 'maps/background.jpg'
20200418 17:48:48 error gui/draw: Image: 'maps/background.jpg' not found and won't be drawn.
Could not create a new surface in get_surface_portion()
failed to make neutral surface
```

**Comment by Pentarctagon** [comment]
A google search for `Could not create a new surface in get_surface_portion()` returns another issue: #2915.  The suggested fix for that was that having Wesnoth be 64-bit on Windows as well would fix the issue, but given that rebhuhn (from the forum thread I linked above) built the 64-bit VS project and is still having this issue, then that means that either this is an unrelated problem or that #2915 won't be fixed by having a 64-bit executable.

A search for `failed to make neutral surface` then turns up [this](https://www.wesnoth.org/forum/viewtopic.php?t=51065) post, with screenshots that look identical to the issue shown in the screenshot here.

**Comment by GaussLM** [comment]
Battle for Wesnoth v1.14.12
Started on Tue Jun 02 19:20:57 2020

Automatically found a possible data directory at E:\SteamLibrary\steamapps\common\wesnoth

Data directory:               E:\SteamLibrary\steamapps\common\wesnoth
User configuration directory: C:\Users\USER\Documents\My Games\Wesnoth1.14
User data directory:          C:\Users\USER\Documents\My Games\Wesnoth1.14
Cache directory:              C:\Users\USER\Documents\My Games\Wesnoth1.14\cache

Setting mode to 1600x900
Checking lua scripts... ok
Failed to create a surface of size 1418x870 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 1418x870 Reason: Out of memory
20200602 20:44:35 error gui/draw: SDL_CreateSoftwareRenderer failed, reason: Can't create renderer for NULL surface
Failed to create a surface of size 1418x870 Reason: Out of memory
20200602 20:44:35 error gui/draw: SDL_CreateSoftwareRenderer failed, reason: Can't create renderer for NULL surface

**Comment by gfgtdf** [comment]
As we suspected it fails with "out of memory" I'd be nice if you could give us more information about when in happens in particular: How much memory does wesnoth use at the time of crash how much ram do you have, are you running a 32 but wesnoth executable? which add-ons are installed? At which campaign did this happen?

**Comment by GaussLM** [comment]
See #4934 for further information.

While I didn't think to check memory usage at time of crash, it seems unlikely to have been just running out of physical memory given I have 32 GB installed.  If/when it happens again I will check though.

Installed add-ons:
Era of Legends
Era of Magic
Era of Magic - Resources
Era of Music
Era of Myths
Era of Myths Mappack

**Comment by AI0867** [comment]
Well, as stated earlier, most (if not all) wesnoth builds for windows are 32 bits, meaning there's only 4 GiB addressable. That's a lot easier to run out of, especially if the OS claims part of it.

**Comment by gfgtdf** [comment]
A common cause of high cpu  usage can the the sheer number of  add-ons installed due to the game config getting to big however considering the small amount of addons installed this doesn't seem to be the case here.

Of course another source are simply menleas with can for example mean objects that are not referenced by any pointers, or caches are that never cleared

What im wondering right now is whether this is windows specific since all reports come from windows, on the other hand windows is also simply the majority of the userbase and maybe most linux builds are also 64 bit.

**Comment by gfgtdf** [comment]
Oh another question: do you play with sound on? IF yes try to disable all sounds in your preferences and check whether it still happens

**Comment by GaussLM** [comment]
Memory at title screen just after starting, prior to loading a file:
![image](https://user-images.githubusercontent.com/66341456/83671942-7afee980-a5a3-11ea-9924-483ab709ab74.png)

Immediately after loading The Long March:
![image](https://user-images.githubusercontent.com/66341456/83672000-9538c780-a5a3-11ea-9f66-c4230e15e113.png)

Was only on the second turn when I got a probably-unrelated error (included here just in case it somehow is related):
![image](https://user-images.githubusercontent.com/66341456/83672484-58210500-a5a4-11ea-9a53-cd4faeb19054.png)
![image](https://user-images.githubusercontent.com/66341456/83672557-7129b600-a5a4-11ea-92fb-2a7bf6bb1708.png)

Was able to hit 'Ignore' and just keep going though.

Played through one scenario, taking my time, then quit to main menu.  Memory usage slowly but steadily crept upward as I played.
Before quitting:
![image](https://user-images.githubusercontent.com/66341456/83677923-e26c6780-a5aa-11ea-9fac-56990f67d85a.png)
After (at Title Screen):
![image](https://user-images.githubusercontent.com/66341456/83677993-fdd77280-a5aa-11ea-9f12-b2fe709e0d20.png)
After hitting F5:
![image](https://user-images.githubusercontent.com/66341456/83678068-1a73aa80-a5ab-11ea-970c-a700f9e99bbd.png)

**Comment by GaussLM** [comment]
I do play with sound on, I will test with it disabled after I see about an unrelated bug in the scenario I found while playing.

**Comment by GaussLM** [comment]
After disabling all sound options in preferences, memory usage still crept up.  At ~1.8 GiB usage I began experiencing visual effects similar to those originally reported in this issue:
![image](https://user-images.githubusercontent.com/66341456/83790231-d85b6f00-a665-11ea-8441-1e570f4f7565.png)

A minute or two later it crashed to desktop without giving any error message.

```
Setting mode to 1600x900
Checking lua scripts... ok
Failed to create a surface of size 1600x900 Reason: Out of memory
20200604 13:16:03 error gui/draw: SDL_CreateSoftwareRenderer failed, reason: Can't create renderer for NULL surface
Failed to create a surface of size 1600x900 Reason: Out of memory
20200604 13:16:03 error gui/draw: SDL_CreateSoftwareRenderer failed, reason: Can't create renderer for NULL surface
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 1213x821 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x500 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 895x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 1213x821 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 1213x821 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x500 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 895x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 1213x821 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 1213x821 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x500 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 895x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 920x730 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 182x900 Reason: Out of memory
failed to make neutral surface
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 1213x821 Reason: Out of memory
Could not create a new surface in get_surface_portion()
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 594x552 Reason: Out of memory
Failed to create a surface of size 596x496 Reason: Out of memory
20200604 13:19:48 error gui/draw: SDL_CreateSoftwareRenderer failed, reason: Can't create renderer for NULL surface
```

**Comment by Pentarctagon** [comment]
@GaussLM If you move all the add-ons to another directory and clear Wesnoth's cache, does the issue still happen?

**Comment by GaussLM** [comment]
Moved add-ons out of Wesnoth's directory, then cleared the cache -- no difference.  Quitting to title screen and clearing/purging cache after memory usage built up to ~1 GiB also had no effect.

Tried a few things while watching memory usage and found that I can drastically speed up the problem by repeatedly loading the game -- went from playing for 2-3 hours before crash down to 2-3 minutes if I do nothing other than constantly reload.

Memory usage was slowly but steadily increasing as I played even when I wasn't doing any loading.

**Comment by gfgtdf** [comment]
> went from playing for 2-3 hours before crash down to 2-3 minutes if I do nothing other than constantly reload.

Ohh thx this is quite helpful, this also reproduces this issue for me ( wesnoth 1.15.3,  ubuntu18 ), i still didnt figure out how to debug here though. But i think its shodul also be rpreoducable for everyone else then.

**Comment by gfgtdf** [comment]
and is also seems to irrelvent which savefile it is, you can use a normal savefile made durign any turn, and constatly use "ctrl+s" "enter"  to raise the memory very quickly.

**Comment by Pentarctagon** [comment]
Same - it didn't crash for me since the linux 1.14 executable is 64-bit, but I was able to keep reloading an AToTB autosave until it got to ~2 GBs RAM usage.  Each reload increased RAM usage by ~15 MBs.

**Comment by gfgtdf** [comment]
further testing shows that the speed in whcih it increses does not depend on the units on the name, but very much on the terrain of the map (a only-grassland maps leads to much less  memory incrementation)


> I was only loading -- "ctrl+o" "enter", no saving involved.

Oh yes, i mistypes i meant ctrl+o

**Comment by Pentarctagon** [comment]
Perhaps images keep being loaded into memory each time a scenario is loaded instead of either using the image cache or being released from memory?

**Comment by gfgtdf** [comment]
yes this could be possible

**Comment by gfgtdf** [comment]
hmm so i tested that when i just use [set_terrain] liek this:
```
local terrains = {"Wwg", "Wst", "Wwt", "Wwf", "Wwrg", "Wwr", "Wwrt", "Ss", "Sm", "Gg", "Gs", "Gd", "Gll", "Rb", "Re", "Rd", "Rr", "Rrc", "Rp", "Ai", "Aa", "Dd", "Ds" }
local c = 0
while true do
	for x=1,10 do
		for y=1,10 do
			c = c + 1
			if c > #terrains then
				c = 1
			end
			wesnoth.set_terrain(x,y, terrains[c])
		end
	end
	wesnoth.redraw {}
end

```
ina  ascenario the memory does not increase

**Comment by Pentarctagon** [comment]
Is there a way to see how many images are in the image cache/how much memory it's using?  I don't really know anything about that code other than what was mentioned as part of #2915.

**Comment by gfgtdf** [comment]
i don't know much about it either, but afaik the image cache should be cleared when a new game is
 started, except for this object though https://github.com/wesnoth/wesnoth/blob/8c6e637b9d153ddb8da9b37ad44d128074076d5f/src/picture.cpp#L155

**Comment by gfgtdf** [comment]
actually the code directly mentions a memoryleak related to that. relates to that in that file

**Comment by Pentarctagon** [comment]
Closing since this looks to be the same as #4934 which has been fixed.
